### DISCUSSION TITLE
What to do about RTSP errors

### DISCUSSION THREAD

**Discussion by ppbrown** [Motion-Project/motionplus#67] [General]
I'm mostly trying to just pull static images from cameras.
Whether or not thats relevant I'm seeing a boatload of errors
(to a Reolink camera)

sample

> [INF][ENC][03:nc03:MyCam1] movie_avcodec_log: RTP: PT=61: bad cseq 6f8d expected=a251
> [INF][ENC][03:nc03:MyCam1] movie_avcodec_log: RTP: PT=61: bad cseq a251 expected=6f8f
> [INF][ENC][03:nc03:MyCam1] movie_avcodec_log: RTP: PT=61: bad cseq 6f8f expected=a253

**Comment by Mr-Dave** [comment]
These are from the ffmpeg libs.  

You may try to change the `netcam_params` and specify a different capture_rate.  Or search the ffmpeg options and try some of those.  

Some forum comments pointed to ffmpeg not reordering the packets (which seems to be the problem showing in your log snippet) and the forum said that someone should work on implementing that in ffmpeg.  But, that comment was long ago so maybe there is an ffmpeg option that can be added to `netcam_params` , maybe not.  

The final option on how to handle them is that you could just edit the motionplus code and just not print them to the log.

**Comment by ppbrown** [comment]
huh.
How does a different capture rate help? What would recommend I aim for?

**Comment by Mr-Dave** [comment]
I do not know that it will make a difference.....the default MotionPlus is 1+ what the camera indicates as its FPS.  But not all cameras report that so it could fall back to the rate in the configuration which could be substantially different.  And different capture rates vs FPS have been observed to cause issues.

**Comment by Mr-Dave** [comment]
I'm curious.  Did slowing the `capture_rate` to be just the camera FPS or FPS minus 1 change the frequency of these messages?

**Comment by ppbrown** [comment]
My experiments indicate best values so far are

    netcam_params capture_rate 16
    framerate 2

It should be noted that motionplus complains it cant detect the FPS rate of
the camera.
(but we usually set them to 15)

**Comment by ppbrown** [comment]
oops. sorry for not being explicit:
with the above settings, I am not seeing ANY of those complaints any more,
usually.

**Comment by ppbrown** [comment]
correction.
now, of course, I'm suddenly seeing a bunch again :-/

**Comment by ppbrown** [comment]
Hmm. additional data. I hadnt configured it properly in my most recent test
bed.
Note to others: when you are running with debug -d 8, you MUST see your
chosen capture rate
 (in this case, 16)
or you didnt do it properly.


[INF][ALL][01:ml01:Cam102] mlp_init: Initialize
[INF][ALL][00:ml00:Cam101] util_parms_parse: Starting parsing parameters
[NTC][VID][01:ml01:Cam102] netcam_start: Opening Netcam
[DBG][ALL][00:ml00:Cam101] util_parms_parse_comma: Parsing:
>capture_rate=16< >0 11 13 14<


that being said, even with the above.. I sometimes see the problem. It all
depends on whether it starts up cleanly or not.
So there's some kind of "initial sync" problem.


Actually, I just checked, and these cams are actually at 7fps.
so, does the value need to be exactly (real-fps+1) ?

Hmm. i guess I'll answer that myself.
setting capture rate to 8, didnt make the problem go away.

**Comment by Mr-Dave** [comment]
I suspect these are all about the capture rate being faster than what the camera is providing.   i.e. The decoder is expecting to get the next in the sequence but with this higher capture rate, it keeps getting sent the same one over and over again. 

I selected the FPS plus one because in my experience if a low capture rate is selected, the camera keeps disconnecting.  Therefore, it is a balance....if the camera is at 7FPS, then set it at 8 and get a few of these messages, set it at 7 and hope things stay in sync, set it at 6 and hope it gets enough data to stay connected...

edit:  I'd note that for my cameras, I do not worry about these messages.  I just set the capture_rate so that it doesn't disconnect and that is more important to me.
